mantle/platform/conf/conf.go

```diff
@@ -398,6 +398,49 @@ func MergeAllV34exp(confObjs []*Conf) (*UserData, error) {
 	return userData, nil
 }
 
+// Config is compressed and added to another via data url
+func ReplaceAndMaybeCompress(config string) (string, error) {
+	originalURL := base64.StdEncoding.EncodeToString([]byte(config))
+	originalSize := len([]byte(originalURL))
+	// Compress config
+	var buff bytes.Buffer
+	writer, err := gzip.NewWriterLevel(&buff, gzip.BestCompression)
+	defer writer.Close()
+	if _, err := writer.Write([]byte(config)); err != nil {
+		return "", err
+	}
+	if err := writer.Close(); err != nil {
+		return "", err
+	}
+	// Encode as data url and add to replace clause in new config
+	url := dataurl.EncodeBytes(buff.Bytes())
+	compressionAlgo := "gzip"
+	newConfigToReplace := v34exptypes.Config{
+		Ignition: v34exptypes.Ignition{
+			Version: "3.4.0-experimental",
```

Comment:
I was working under the assumption that the config being replaced must be as new or newer than the config in the replace clause. Since kola tests are allowed to use `3.4.0-exp` configs, I thought this was needed to ensure that the condition is always satisfied. If that's not the case, it makes sense to use the latest stable spec instead.

Similar reason for the use of the experimental spec in the preceding function. Since a non-exclusive test is not restricted from using `3.4.0-exp`, the experimental spec may be present in the list of configs being merged. Consequently, the result will need to be the experimental spec as well.

Reply:
> I was working under the assumption that the config being replaced must be as new or newer than the config in the replace clause.

Nope.  Ignition internally converts all configs to the experimental version before merging them.